server/datastore/cached_mysql/cached_mysql.go

```diff
@@ -62,13 +130,15 @@ func (ds *cachedMysql) NewAppConfig(ctx context.Context, info *fleet.AppConfig)
 
 	ds.c.Set(appConfigKey, ac, defaultAppConfigExpiration)
 
-	return ac.Clone()
+	return ac, nil
```

Comment:
Why not return `clone(ac)`? (otherwise the caller of `NewAppConfig` could modify the in-cache item `ac`)
(The same in other places where you `Set` on the cache and then return the item.)

Reply:
This is true. Initially, I thought that doing it on `(*cache.Cache).Get` would be enough. But you are right, we should also do it on `(*cache.Cache).Set` just to be safe. Added cloning in `(*cloneCach).Set` in 98456d25.